snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedEditingContext.java

```diff
@@ -619,126 +628,96 @@ public SnomedDeletionPlan canDelete(Concept concept, SnomedDeletionPlan deletion
 		 */
 		
 		for (Description description : concept.getDescriptions()) {
-			deletionPlan = canDelete(description, deletionPlan, force);
+			canDelete(description, force);
 		}
 		
 		for (Relationship outboundRelationship : concept.getOutboundRelationships()) {
-			deletionPlan = canDelete(outboundRelationship, deletionPlan, force);
+			canDelete(outboundRelationship, force);
 		}
 		
 		SnomedRefSet refSet = lookupIfExists(concept.getId(), SnomedRefSet.class);
 		if (refSet != null) {
-			deletionPlan = canDelete(refSet, deletionPlan, force);
+			canDelete(refSet, force);
 		}
 		
 		List<SnomedRefSetMember> referringMembers = refSetEditingContext.getReferringMembers(concept);
 
-		// If this concept is a member of the description format reference set, descriptions of this type have to be updated
-		for (SnomedRefSetMember member : referringMembers) {
-			if (REFSET_DESCRIPTION_TYPE.equals(member.getRefSetIdentifierId())) {
-				for (SnomedDescriptionIndexEntry entry : getRelatedDescriptions(member.getReferencedComponentId())) {
-					final Description description = lookup(entry.getId(), Description.class);
-					if (null == description) {
-						throw new SnowowlRuntimeException("Description does not exist in store with ID: " + entry.getId());
-					} else {
-						deletionPlan.addDirtyDescription(description);
-					}
-				}
+		Optional<SnomedRefSetMember> isDescriptionType = referringMembers.stream()
+				.filter(member -> member.isActive() && member.getRefSetIdentifierId().equals(REFSET_DESCRIPTION_TYPE)).findAny();
+		
+		if (isDescriptionType.isPresent()) {
+			
+			boolean hasRelatedDescriptions = SnomedRequests.prepareSearchDescription()
+				.setLimit(1)
```

Comment:
You can safely limit the result set to zero. The returned collection resource will still have a total number reflecting the total number of matches.